src/field/pokemon.ts

```diff
@@ -1372,6 +1372,10 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
       typeMultiplier.value = 0;
     }
 
+    if (this.getTag(TarShotTag) && (this.getMoveType(move) === Type.FIRE)) {
```

Comment:
```suggestion
    if (this.hasTag(TarShotTag) && (this.getMoveType(move) === Type.FIRE)) {
```
I think in this case you would use `hasTag()`. Same result but clearer on whats' going on

Reply:
we do not have `hasTag` for `Pokemon` object